src/NuGet.Clients/NuGet.PackageManagement.UI/Models/ItemsChangeObservableCollectionBase.cs

```diff
@@ -13,9 +14,23 @@ namespace NuGet.PackageManagement.UI
     /// <typeparam name="T"></typeparam>
     public class ItemsChangeObservableCollection<T> : ObservableCollection<T>
     {
+        public ItemsChangeObservableCollection()
+            : base() { }
+
+        public ItemsChangeObservableCollection(IEnumerable<T> collection)
+            : base(collection) { }
         public void Refresh()
         {
             OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
         }
+
+        public void AddRange(IEnumerable<T> newItems)
+        {
+            foreach (var item in newItems)
```

Comment:
Line 29-32 can be replaced by simply `Items.AddRange(newItems);`, .net has quite many useful built-in functions.

Reply:
`Items` is an `IList`, which does not specify `AddRange`. If you're seeing an `AddRange`, it's probably coming from some NuGet extension methods, and not from .NET. Only `List<T>` specifies an `AddRange`. See https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.collection-1.items?view=net-6.0#system-collections-objectmodel-collection-1-items